packages/crypto/src/keys/index.ts

```diff
@@ -79,7 +76,7 @@ export async function generateKeyPairFromSeed (type: KeyTypes, seed: Uint8Array,
 /**
  * Converts a protobuf serialized public key into its representative object
  */
-export function unmarshalPublicKey (buf: Uint8Array): PublicKey {
+export function unmarshalPublicKey (buf: Uint8Array): PublicKey<KeyTypes> {
```

Comment:
Should we?

```suggestion
export function unmarshalPublicKey <T extends KeyTypes>(buf: Uint8Array): PublicKey<T> {
```

Reply:
Maybe, but I think we might want to constrain it to the cryptographic types?